pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd *cobra.Command, args []string) erro
 		// Set the specified repo as the only one to output dependencies for
 		repos[repo] = tags
 	} else {
-		// Get all image repositories from build configurations
-		repos = getRepos(buildConfigList)
-
-		// Make sure from now on that the --all flag is true
-		// since we are building dependency trees for every
-		// image repository available either in the current
-		// namespace or in every namespace
-		all = true
+		// all the tags of all the imagestreams across all the namespaces
+		for _, ns := range namespaces {
```

Comment:
Right, my point is its going from o(n) to o(2n) not from o(1).  If we can't make calls per namespace, then we were already broken.  


From: Michail Kargakis <notifications@github.com>
Sent: Apr 23, 2015 5:53 PM
To: openshift/origin
Cc: Ben Parees
Subject: Re: [origin] add imagestream refs for custom and docker strategies (#1693)

> @@ -186,14 +189,24 @@ func RunBuildChain(f *clientcmd.Factory, cmd *cobra.Command, args []string) erro
>  		// Set the specified repo as the only one to output dependencies for
>  		repos[repo] = tags
>  	} else {
> -		// Get all image repositories from build configurations
> -		repos = getRepos(buildConfigList)
> -
> -		// Make sure from now on that the --all flag is true
> -		// since we are building dependency trees for every
> -		// image repository available either in the current
> -		// namespace or in every namespace
> -		all = true
> +		// all the tags of all the imagestreams across all the namespaces
> +		for _, ns := range namespaces {

We have to get the build configs in any case. 

---
Reply to this email directly or view it on GitHub:
https://github.com/openshift/origin/pull/1693/files#r29007517